app/addons/documents/revisionTree/component.react.jsx

```diff
@@ -0,0 +1,330 @@
+
+              if (map[rev]) {
+                x = map[rev][0];
+                y = map[rev][1];
+              }
+
+              var lineObj = {
+                "x" : x,
+                "y" : y,
+                "nextX" : nextX,
+                "nextY" : nextY
+              };
+
+              lineObjs.push(lineObj);
+            }
+            if (map[rev]) {
+              break;
+            }
+            maxX = Math.max(x, maxX);
+            maxY = Math.max(y, maxY);
+            levelCount[pos]++;
+
+            node(x, y, rev, isLeaf, rev in deleted, rev === winner, minUniq);
+            map[rev] = [x, y];
+          }
+        }
+        paths.forEach(drawPath);
+      };
+
+      var minUniqueLength = function(arr){
+       function strCommon(a, b){
+        if (a === b) return a.length;
+        var i = 0;
+        while(++i){
+          if(a[i - 1] !== b[i - 1]) return i;
+        }
+      }
+
+      var array = arr.slice(0);
+      var com = 1;
+      array.sort();
+      for (var i = 1; i < array.length; i++){
+        com = Math.max(com, strCommon(array[i], array[i - 1]));
+      }
+      return com;
+    };
+
+    function node(x, y, rev, isLeaf, isDeleted, isWinner, shortDescLen){
+      circ(x, y, r, isLeaf, isDeleted, isWinner);
+      var pos = rev.split('-')[0];
+      var id = rev.split('-')[1];
+      var opened = false;
+
+      var textObj = {
+        "stLeft": (x-40) + "px",
+        "stTop": (y-30) + "px",
+        "short": pos + '-' + id.substr(0, shortDescLen),
+        "long": pos + '-' + id
+      };
+
+      textObjs.push(textObj);
+    }
+
+    var circ = function(x, y, r, isLeaf, isDeleted, isWinner) {
+
+      var leafStat = "";
+
+      if (isLeaf) {
+        leafStat = "leaf";
+      }
+      if (isWinner) {
+        leafStat = "winner";
+      }
+      if (isDeleted) {
+        leafStat = "deleted";
+      }      
+
+      var nodeObj = {
+        "x" : x,
+        "y" : y,
+        "class" : leafStat
+      };
+
+      nodeObjs.push(nodeObj);
+    };
+
+    var App = React.createClass({
+      getInitialState: function() {
+        return {
+          lines: [],
+          treeNodes: [],
+          nodeTextObjs: []
+        };
+      },
+
+      componentDidMount: function() {
+        var result = [];
+        var paths = [];
+        var deleted = {};
+        var optionsVal = store.getTreeOptions();
+        // var winner = this.props.data.winnerRev;
+        var winner = optionsVal.winnerRev;
+        // var url = '/'+this.props.data.db+'/'+this.props.data.docID;
+        var url = '/'+optionsVal.db+'/'+optionsVal.docID;
+        // var winner = this.props.data.;
+        var minUniq = 0;
+        $.get(app.host+url+'?open_revs=all&revs=true', function(rslt) {
+          var data = rslt;
+          if (this.isMounted()) {
+            var x = data.split(/(\n|\r\n|\r)/);
+
+            for (var i = 0; i <x.length; i++) {
+              if(String(x[i]).charAt(0) == "{"){
```

Comment:
Once we get down to the nitty gritty, we'll want to get all the code style consistent with the rest of the codebase, like here: space after the if, triple equals, not double (we always try to use strict equality). But I won't bore you with picking out all the instance in the code - just read over the style doc and get to it whenever you refactor. No rush!